tools/mkclass/classcompiler.cpp

```diff
@@ -376,14 +376,16 @@ void ClassCompiler::HandleClass(const Klass& klass, const ClassDebugInfo&)
 		<< "}" << std::endl << std::endl;
 
 	/* GetLoadDependencies */
-	m_Header << "\t" << "std::vector<String> GetLoadDependencies() const override;" << std::endl;
+	m_Header << "\t" << "const std::unordered_set<Type*>& GetLoadDependencies() const override;" << std::endl;
 
-	m_Impl << "std::vector<String> TypeImpl<" << klass.Name << ">::GetLoadDependencies() const" << std::endl
+	m_Impl << "const std::unordered_set<Type*>& TypeImpl<" << klass.Name << ">::GetLoadDependencies() const" << std::endl
 		<< "{" << std::endl
-		<< "\t" << "std::vector<String> deps;" << std::endl;
+		<< "\t" << "static const std::unordered_set<Type*> deps ({" << std::endl;
 
 	for (const std::string& dep : klass.LoadDependencies)
-		m_Impl << "\t" << "deps.emplace_back(\"" << dep << "\");" << std::endl;
+		m_Impl << "\t\t" << dep << "::TypeInstance.get()," << std::endl;
```

Comment:
Looks like this breaks non-unity builds, at least I get this error when building locally:

```
In file included from .../lib/remote/endpoint.cpp:4:
.../cmake-build-debug/lib/remote/endpoint-ti.cpp: In member function ‘virtual const std::unordered_set<icinga::Type*>& icinga::TypeImpl<icinga::Endpoint>::GetLoadDependencies() const’:
.../cmake-build-debug/lib/remote/endpoint-ti.cpp:172:23: error: incomplete type ‘icinga::Zone’ used in nested name specifier
  172 |                 Zone::TypeInstance.get(),
      |                       ^~~~~~~~~~~~
.../cmake-build-debug/lib/remote/endpoint-ti.cpp:173:10: error: no matching function for call to ‘std::unordered_set<icinga::Type*>::unordered_set(<brace-enclosed initializer list>)’
  173 |         });
      |          ^
```

But given that the builds on GH Actions work, this should be fixable by adding includes. However, if this requires larger changes to `mkclass`, I'm fine with keeping `GetByName()` as this done only once anyways.

Reply:
This change shouldn’t require us to manually include the load deps.
Also we can't autogen them.
And in this special case I'm afraid it's a chicken-and-egg problem (Zone/Endpoint).